### DISCUSSION TITLE
Can agents created through the AutoGen framework interact with agents not created by that framework

### DISCUSSION THREAD

**Discussion by henrynn** [microsoft/autogen#4458] [Q&A]
1. Can agents created through the AutoGen framework interact with agents not created by that framework?
2. Task planning is on the cloud side, plugins are on the edge side, and execution may be on the edge side; can this architecture be adapted?

**Comment by ekzhu** [comment]
it is possible currently with the Core API. See examples of langchain and llama index agents:

- https://microsoft.github.io/autogen/dev/user-guide/core-user-guide/cookbook/langgraph-agent.html
- https://microsoft.github.io/autogen/dev/user-guide/core-user-guide/cookbook/llamaindex-agent.html

> Task planning is on the cloud side, plugins are on the edge side, and execution may be on the edge side; can this architecture be adapted?

Can you take a look at distributed agent runtime to see if that meets your requirements?

1. https://microsoft.github.io/autogen/dev/user-guide/core-user-guide/core-concepts/architecture.html#distributed-agent-runtime
2. https://microsoft.github.io/autogen/dev/user-guide/core-user-guide/framework/distributed-agent-runtime.html
